### DISCUSSION TITLE
Refused to set unsafe header "cookie"

### DISCUSSION THREAD

**Discussion by scottquested** [mswjs/msw#818] [Q&A]
We are getting this error `Refused to set unsafe header "cookie"` only in chrome

<img width="762" alt="Screenshot 2021-07-19 at 11 37 21" src="https://user-images.githubusercontent.com/4488112/126147636-915602f0-ee88-4b68-a1f3-e9101b2e3b80.png">
<img width="740" alt="Screenshot 2021-07-19 at 11 36 55" src="https://user-images.githubusercontent.com/4488112/126147642-91705663-4462-4555-a5e9-984bceed9a6b.png">

Thought I would ask here first before opening an issue

**Comment by kettanaito** [comment]
Hey, @scottquested.

As the error suggests, it looks like you're setting the `"cookie"` header on your mocked response somewhere:

```js
return res(ctx.set('cookie', 'value'))
```

If that's the case, please use [`ctx.cookie()`](https://mswjs.io/docs/recipes/cookies#mock-response-cookies) to set a cookie on a mocked response. That utility is designed to emulate response cookie functionality without compromising the security behind the `cookie` header. 

If that's not the case, could you please share your request handlers setup and describe in more detail _when_ exactly this error happens (what kind of request are you making, what handlers do you have, do you expect this response to be mocked). Ideally, to resolve this issue quicker, I'd ask you for a minimal reproduction repository.

**Reply by scottquested** [reply]
Thanks for your prompt reply @kettanaito it's not actually happening on a mocked response. This happens even when we have no mocked responses. 

```js
// Browser file
import { setupWorker } from 'msw';
import { devHandlers } from './handlers';
// This configures a Service Worker with the given request handlers.
export const worker = setupWorker(...devHandlers);
```

```js
// devHandlers file
import { rest } from 'msw';
export const devHandlers = [];
```

```js
// App index file
if (
      process.env.NODE_ENV === 'development' &&
      process.env.REACT_APP_MOCK_API === 'true'
    ) {
      // eslint-disable-next-line global-require
      const { worker } = require('./mocks/browser');
      worker.start({
        warnOnUncaptured: false,
        waitUntilReady: true,
        onUnhandledRequest: () => {
          return false;
        },
      });
    }
```

**Reply by scottquested** [reply]
If we comment out the options it seems to run this error on each API that isn't mocked...
```js
if (
      process.env.NODE_ENV === 'development' &&
      process.env.REACT_APP_MOCK_API === 'true'
    ) {
      // eslint-disable-next-line global-require
      const { worker } = require('./mocks/browser');
      worker.start({
        // warnOnUncaptured: false,
        // waitUntilReady: true,
        // onUnhandledRequest: () => {
        //   return false;
        // },
      });
    }
```
<img width="716" alt="Screenshot 2021-07-19 at 13 11 39" src="https://user-images.githubusercontent.com/4488112/126158120-f9ab4921-5929-4062-af40-f0cd7368efd7.png">

**Reply by scottquested** [reply]
Chrome `Version 91.0.4472.114 (Official Build) (x86_64)`

**Comment by scottquested** [comment]
Ok, we have found the issue. Because we are running locally non-https (in fallback mode) this was causing it. Perhaps cookies should not be set in fallback mode in Chrome. We added our url to the flag `chrome://flags/#unsafely-treat-insecure-origin-as-secure` which by passes this.

**Reply by kettanaito** [reply]
I think it's a valid point that bypassed XHR requests must not set the forbidden cookies. What's complicated is that it's the case only for the browser. As far as I know there's no such issue in Node.js. Definitely calls for a test in the "interceptors" library.

**Reply by kettanaito** [reply]
Opened an issue in the interceptors repo: https://github.com/mswjs/interceptors/issues/141
